src/routing/lazy_messaging.rs

```diff
@@ -97,9 +103,9 @@ fn create_other_section_message(
     node: &Node,
     section: &Section,
     network: &Network,
+    variant: Variant,
     dst: DstLocation,
-    nonce: MessageHash,
-    // dst_key: Option<bls::PublicKey>,
+    // nonce: MessageHash,
 ) -> Result<Message> {
     let dst_knowledge = dst
```

Comment:
better have `dst_knowledge` as an input parameter.
as now knowledge is updated individually. so our record only reflects the most advanced one.